streams/src/main/java/org/apache/kafka/streams/state/internals/ChangeLoggingSessionBytesStore.java

```diff
@@ -79,15 +91,34 @@ public void init(final StateStoreContext context, final StateStore root) {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public void remove(final Windowed<Bytes> sessionKey) {
+        if (context.recordMetadata().isPresent()) {
+            final RecordMetadata meta = context.recordMetadata().get();
+            position = position.update(meta.topic(), meta.partition(), meta.offset());
+        }
         wrapped().remove(sessionKey);
-        context.logChange(name(), SessionKeySchema.toBinary(sessionKey), null, context.timestamp());
+        Optional<Position> optionalPosition = Optional.empty();
+        if (consistencyEnabled) {
+            optionalPosition = Optional.of(position);
+        }
+        context.logChange(name(), SessionKeySchema.toBinary(sessionKey), null, context.timestamp(), optionalPosition);
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public void put(final Windowed<Bytes> sessionKey, final byte[] aggregate) {
+        if (context.recordMetadata().isPresent()) {
+            final RecordMetadata meta = context.recordMetadata().get();
+            position = position.update(meta.topic(), meta.partition(), meta.offset());
+        }
         wrapped().put(sessionKey, aggregate);
-        context.logChange(name(), SessionKeySchema.toBinary(sessionKey), aggregate, context.timestamp());
+        Optional<Position> optionalPosition = Optional.empty();
+        if (consistencyEnabled) {
+            optionalPosition = Optional.of(position);
+        }
+        context.logChange(
+                name(), SessionKeySchema.toBinary(sessionKey), aggregate, context.timestamp(), optionalPosition);
```

Comment:
Yeah, I think these will all become just one-line changes.

```suggestion
        context.logChange(name(), SessionKeySchema.toBinary(sessionKey), aggregate, context.timestamp(), position);
```